### PR TITLE
cloudfoundry-cli: 7.0.1 -> 7.2.0, switch to buildGoModule, relocate

### PR DESCRIPTION
**Pull request by risicle**
###### Motivation for this change
Mostly just a bump. Upstream has switched to modules, allowing me to use `buildGoModule` and simplify the build quite a bit. Also moved the package to `pkgs/applications/networking/cluster` because it had been bugging me for ages being in `pkgs/development/tools`.

###### Things done

- [x] Tested using sandboxing ([nix.useSandbox](https://nixos.org/nixos/manual/options.html#opt-nix.useSandbox) on NixOS, or option `sandbox` in [`nix.conf`](https://nixos.org/nix/manual/#sec-conf-file) on non-NixOS linux)
- Built on platform(s)
   - [ ] NixOS
   - [x] macOS
   - [x] other Linux distributions
- [ ] Tested via one or more NixOS test(s) if existing and applicable for the change (look inside [nixos/tests](https://github.com/NixOS/nixpkgs/blob/master/nixos/tests))
- [x] Tested compilation of all pkgs that depend on this change using `nix-shell -p nixpkgs-review --run "nixpkgs-review wip"`
- [x] Tested execution of all binary files (usually in `./result/bin/`)
- [ ] Determined the impact on package closure size (by running `nix path-info -S` before and after)
- [ ] Ensured that relevant documentation is up to date
- [ ] Fits [CONTRIBUTING.md](https://github.com/NixOS/nixpkgs/blob/master/.github/CONTRIBUTING.md).

### CODE REVIEW

File: pkgs/applications/networking/cluster/cloudfoundry-cli/default.nix [hunk 1]
```diff
@@ -24,24 +25,16 @@ buildGoPackage rec {
...
+    "-s"
+    "-w"
+    "-X code.cloudfoundry.org/cli/version.binaryBuildDate=1970-01-01"
+    "-X code.cloudfoundry.org/cli/version.binaryVersion=${version}"
+  ];
 
-  installPhase = ''
-    install -Dm555 out/cf "$out/bin/cf"
+  postInstall = ''
+    mv "$out/bin/cli" "$out/bin/cf"
```
Comment (SuperSandro2000) [inline pkgs/applications/networking/cluster/cloudfoundry-cli/default.nix:37]:
```suggestion
    mv $out/bin/cli $out/bin/cf
```

**Review by SuperSandro2000 — CHANGES_REQUESTED** [review]

File: pkgs/applications/networking/cluster/cloudfoundry-cli/default.nix (hunk 1 above)
Comment (risicle) [reply pkgs/applications/networking/cluster/cloudfoundry-cli/default.nix:37]: Jesus you are just exhausting

Comment (SuperSandro2000) [reply pkgs/applications/networking/cluster/cloudfoundry-cli/default.nix:37]: Maybe this is not strictly necessary.

**Comment by SuperSandro2000** [comment]
This is a semi-automatic executed nixpkgs-review with [nixpkgs-review-checks extension](https://github.com/SuperSandro2000/nixpkgs-review-checks). It is checked by a human on a best effort basis and does not build all packages (e.g. lumo, tensorflow or pytorch).
If you have any questions or problems please reach out to SuperSandro2000 on IRC.

Result of `nixpkgs-review pr 111482` run on x86_64-darwin [1](https://github.com/Mic92/nixpkgs-review)
<details>
<summary>
1 package built:
</summary>
<ul>
<li>cloudfoundry-cli</li>
</ul>
</details>

**Pull request merged**